src/services/rate/client/chain.ts

```diff
@@ -0,0 +1,27 @@
+import {IRateService} from "../rate.service.interface";
+
+export interface IChain extends IRateService{
+    setNext(next: IChain): void;
+}
+
+export class BaseChain implements IChain {
+    private next: IChain | null
+
+    constructor(private rateService: IRateService) {}
+
+    setNext(next: IChain): IChain {
+        this.next = next;
+        return next;
+    }
+
+    async getExchangeRate(): Promise<number> {
+        try {
+            return await this.rateService.getExchangeRate();
+        } catch (error) {
+            if (this.next === null) {
```

Comment:
you don't need to check if some variable is null explicitly you can write just like this 
```js
if (!this.next) { throw error }
```